private/bufpkg/bufapimodule/module_reader.go

```diff
@@ -61,15 +61,15 @@ func (m *moduleReader) GetModule(ctx context.Context, modulePin bufmoduleref.Mod
 		// malformed pin
 		return nil, err
 	}
-	resp, err := m.download(ctx, modulePin)
-	if err != nil {
-		return nil, err
-	}
 	identityAndCommitOpt := bufmodule.ModuleWithModuleIdentityAndCommit(
 		moduleIdentity,
 		modulePin.Commit(),
 	)
 	if m.tamperProofingEnabled {
```

Comment:
Curious, when is this set? I assume whenever `moduleReader` is instantiated. If user sets/unsets the ENV value, does this get updated?

Reply:
This is from a previous PR but it gets set here: https://github.com/bufbuild/buf/blob/7d03938f0f3da7d3d2c12402c87df8f89f376cd2/private/bufpkg/bufapimodule/module_reader.go#L34-L39

When re-running the CLI, it will get the current value of the ENV value. It has no support for re-checking an env var set via https://pkg.go.dev/os#Setenv if that's what you are asking.

Reply:
> When re-running the CLI, it will get the current value of the ENV value.

You mean every time I run any `buf` command?

Reply:
> You mean every time I run any `buf` command?

Yep!